From patchwork Tue Sep  6 19:48:53 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12968101
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id EF1C6C54EE9
	for <linux-mm@archiver.kernel.org>; Tue,  6 Sep 2022 19:50:26 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 01E658D0012; Tue,  6 Sep 2022 15:50:00 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id D57B78D0009; Tue,  6 Sep 2022 15:49:59 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 93DD58D000C; Tue,  6 Sep 2022 15:49:59 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0014.hostedemail.com
 [216.40.44.14])
	by kanga.kvack.org (Postfix) with ESMTP id 5C87B8D000A
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 15:49:59 -0400 (EDT)
Received: from smtpin31.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay04.hostedemail.com (Postfix) with ESMTP id 380AB1A08AC
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:49:59 +0000 (UTC)
X-FDA: 79882701318.31.0D9C76E
Received: from mx0a-00069f02.pphosted.com (mx0a-00069f02.pphosted.com
 [205.220.165.32])
	by imf14.hostedemail.com (Postfix) with ESMTP id CF069100069
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:49:58 +0000 (UTC)
Received: from pps.filterd (m0246617.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 286IdYZe029954;
	Tue, 6 Sep 2022 19:49:40 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=FuNmZU+UMH7Y8mw5oBbNoLDQmSkii/s9w06bWHod278=;
 b=hJAvIoXoWtIMJ691vN3PM4a5u9Dx/VsQZsLObedr+pBeNQ27IcZSNXEsEeKB01DxCskq
 ZJIphcdKNZXv0FpNogdVzPSvf5a+cir80FSnc7149ySS6QVQ/7C7kH4R84Vz4MKXOp8D
 MaS1kRcs0/C5Ldgho0rJbORWgmvleE8I8XDKadpRTVaksAT5NTaGG9sywVSSYS58r8SC
 MzBatpp7Ve0L4TgePEjvuWO+FAfdrEC7vuqZu7ZaHAi/huAFBgn8onSsf1ayHc6HZQm7
 JF1DTbB4wjkCKKzZXRH8SuEj//0tqImDXY5zXMJyT2OPBAYUdzp/MvfAgjurJOZwUG40 PA==
Received: from iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com
 (iadpaimrmta03.appoci.oracle.com [130.35.103.27])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3jbyftpvcs-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:49:40 +0000
Received: from pps.filterd
 (iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com [127.0.0.1])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 286JLGjp039722;
	Tue, 6 Sep 2022 19:49:38 GMT
Received: from nam12-dm6-obe.outbound.protection.outlook.com
 (mail-dm6nam12lp2173.outbound.protection.outlook.com [104.47.59.173])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (PPS) with ESMTPS id
 3jbwc9m7h2-4
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:49:38 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=Dpp7ZRIcSPVuL4EMa8/EChzZlQ90vXtiev0hesBlqvCs6Gr4cLMKtCuExQZn5+jnYkgp9aZUYVAhgeA70QgoXp29ycJ92R6WhhARca++m5mjoaZ3eiyFRO+JJVxVY+mdGTP/35+u1GwO5sfXwgXLbmKuuLW+42kgXNfSuD/8LfPvKAryvpHVT1nSg74cioq3wqD7PZvxxz0v8n5BKv9NIgjF3u4hIcHwFb9BVJp/py0oEX+WUWHFpJiNA9TdzZW7aSfU7g3F8ZWUkvoPPdaz/YXL/arzVhlJs0D/yVFPXGfwiSBSnpXPuPV5ktk6HTR/e1ONXvoUjENCoQY2DxuDOw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=FuNmZU+UMH7Y8mw5oBbNoLDQmSkii/s9w06bWHod278=;
 b=ob0XOVWrfxFxgHoL6G0YYCJlrCStPqmvOhMUfvC+d5cjdwEbLas72IEbp3L4mG+hoqQeDHgBPXK3Iqpf3mypgSDks10Chgu0LSdjtzJjEnTsU0TEUlxVXpy1gMEO3DG9dMEVcE2F9Sfn3jESA1WTLsmtXpIH6Q21uk2ycRiKChvTaT15xnp3sPFcjQ2Z2m6k+rk+/PyqyKxMVlB4Io1BJMCIZ+pe5Pp+CkpcGaXpC4jh+NCdcXKejfudxdUmXWRIyOuhw5ZVXbCIA9B7itICQ5IyLF6FC+lzPZbAn1zC+2JXulaWgFyGO6H8Om6YmCqITicwMGc1wB0J9kRbZvNtdw==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=FuNmZU+UMH7Y8mw5oBbNoLDQmSkii/s9w06bWHod278=;
 b=BP/FjB1BbHC2YqXNbXGbPJ3s8GD4yOuoE4F4UDflPMmIS9n+u7iI+hOVl3UutGz3ZGfZJkNQjEfHCngJZOtzQHEkU33gUTZesvWm+6JQrfW8AfyqdUYBh7U9siEegoeITvdWTCfsBXqmG4DsN+qB2b+UhbB9JXPT+2o+TmyuahQ=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by SJ0PR10MB4670.namprd10.prod.outlook.com (2603:10b6:a03:2dc::12) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5588.10; Tue, 6 Sep
 2022 19:49:35 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223%5]) with mapi id 15.20.5588.018; Tue, 6 Sep 2022
 19:49:35 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>
CC: "Matthew Wilcox (Oracle)" <willy@infradead.org>,
        Michael Ellerman
	<mpe@ellerman.id.au>,
        Benjamin Herrenschmidt <benh@kernel.crashing.org>,
        Paul
 Mackerras <paulus@samba.org>,
        Christophe Leroy <christophe.leroy@csgroup.eu>,
        Liam Howlett <liam.howlett@oracle.com>,
        Vlastimil Babka <vbabka@suse.cz>, Davidlohr Bueso <dave@stgolabs.net>
Subject: [PATCH v14 33/70] powerpc: remove mmap linked list walks
Thread-Topic: [PATCH v14 33/70] powerpc: remove mmap linked list walks
Thread-Index: AQHYwimx5xIJItGJRk2mcitGBwUYWw==
Date: Tue, 6 Sep 2022 19:48:53 +0000
Message-ID: <20220906194824.2110408-34-Liam.Howlett@oracle.com>
References: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 53359466-70b5-4bf3-a057-08da9040ed39
x-ms-traffictypediagnostic: SJ0PR10MB4670:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 LwK1sYWBBfO0ciFwPsAFkHTyzTORUPBd1ZnL+MAWNPTTi2TG7U3wGL73g9uWTt9LbBiXnDKmv3XlFxknPj5H3bkKXv0JV22eUSeylSFM/whaobF2uUa41qfG568BUip6ucJ0zxXB6OMS5V9FU7RNCDCgNQfKtfcicPc+4JG/65sYe/aINRPqL/f68phDySTdPrq/FIcdpJDsFnAl5e8agNqLLKqywhTNQdJbo8+wpcDi8TWkx4tUhm/4DL7JUOj25hoiAvQ5gXxCvOJabLJjbHM1kARkG02QyNmkwSBkA8UcxLSjBiBwLvqMSftqGDlUBS64/BD0xK9XFGrumIGMjSlLytGlRh3bRo7x0NdNbCY21Ch+01Tj15XvwOY7XlnSDjXTGCQ1ekyUT9CXW934fMyt5slv25Ch5PMnhNk2Oie4EJTXLLkOKIkwqdDERbcUv4ScMwcIRcfzz0V7c32EglGyEd/KHpCqBihEt7ivfrql+GHbm8weXVZm5qlTr2M5sgL51axmPqVbGPsPNhRu1yLxVIBKK4YRITPLY9JZ8HUIKh+DUd3gnCl04tjpd+UXGrXUXRINP/F00pFL/7cnC96noNA2Nh6wf8PI4SqLlZp9zSkih0AEZ7VhMpVBLq2f7GwwlRcjEjaDBug5aFvl7i+LrQsEnwYXg9gQDhph3u3ntSsBN0gsfW/FKgs4QHgJefbKPYDh/I8tljd1qs8h5wVQwuKLMMoKyQpbbd4n57486bdhQZMRdgkx7k58wKtdglDOwA6XJKFvRwLygFhXfA==
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(366004)(39860400002)(376002)(346002)(396003)(136003)(6486002)(83380400001)(6512007)(2906002)(26005)(6506007)(66446008)(186003)(122000001)(1076003)(4326008)(66556008)(66476007)(2616005)(76116006)(64756008)(66946007)(8676002)(91956017)(478600001)(36756003)(44832011)(8936002)(54906003)(316002)(86362001)(41300700001)(7416002)(5660300002)(38070700005)(110136005)(71200400001)(6666004)(38100700002);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?q6MLiFPdglHklmPsRn/oCcM?=
	=?iso-8859-1?q?c5tVIpglrqlKskSy1J7jH4QQuxcYT1Zr+4vQT8m+NpHKKmWzYiHwptsK70mm?=
	=?iso-8859-1?q?XbdXdPDCRbUfLS5z8FmMojh6Hz26gDOqKGEivL2uEZkIvloXFIadck4+WAUO?=
	=?iso-8859-1?q?ZFG5jgTZGimLs+tCI1BcgpxkWKbHHx4BSBfw4DE//YmGCi4GZWkKMpWRFmR9?=
	=?iso-8859-1?q?w13tZ5/XwV5eUmFI3bgv6aEV3vsmZv7gbx5rMFzOJZ2bmDCpxtl2PyU06dty?=
	=?iso-8859-1?q?NgmzcMHmkOTbTJk8RTD2Lxk5tAyePaBA0kpm1Qas7kgba2a17ZEP1/CMOLxS?=
	=?iso-8859-1?q?IfGCvrES697JU69uH35+eTUTqSZhHtKTvmRmmkFc8q+kLjLDwW95gu8OWOXv?=
	=?iso-8859-1?q?XK4Olqr/7xHMgwpXEPqSalvzSykkQb+GEq2p13FgWwMhK0TF2fYHby/wUSAI?=
	=?iso-8859-1?q?VOA1reg+9vOfOPeAdoxDQiXSmeL1bOXxvOEX1GMJZXIydidcb4zn1bD2rdZb?=
	=?iso-8859-1?q?2YZCxcs9PlbMAma84scqh4SNd8A7UpmSeI4o0BfkGKgnJCZdFbqDjH5z748z?=
	=?iso-8859-1?q?kjYq/e7RzIHlQvFDf3WNyB9j4faO1m9tE9OO4+8lXc8bF6NMBv2bjSJc5Jdv?=
	=?iso-8859-1?q?EwPbZpwWpcEZmZM4ACDGbPdrmP7d5XiLU/TXTHsydIYbYNZ7v+pP5NPekWkt?=
	=?iso-8859-1?q?4SOInXogqQ1/rOpvf1dDEBKh4ctWdpBQ9xuZNn6A0p7B3ZW7cr+RnwIgDjH0?=
	=?iso-8859-1?q?RlUg+IMKMLFWVRlZXjM1jR4+sKLFFm6AK+FkBpEH8E8QPVlU7drGNhFpTceW?=
	=?iso-8859-1?q?JCOyuLGwutEevP6C8/8yEvYRhqLAGX1m8jxIOm06p5EYzynAj5pP3p0H7Pfu?=
	=?iso-8859-1?q?H9Aoepjf9rrklnAPbyjw7h3kWQTb+kIujGwOVwQZiuukBsYBqHD0tNUWmZs9?=
	=?iso-8859-1?q?HmILvQUdY0oQsf0dvxFQZtXXv27DNrZa4JiMpe3/YhKsojn6IutU/zXdPPBO?=
	=?iso-8859-1?q?l9yjb3ns+u4cLrbuyRnLp9hLf+ShTtqcvTSlsML7uUYhLLFRgbIPziEqHh2i?=
	=?iso-8859-1?q?SMY++VwBnBEUKOiKFb6OzAh0SPEzkE1Cpi5fw6BBPS7AIYTI0Dhn8YxsjkiH?=
	=?iso-8859-1?q?3wQ+4icBd02EZDjVSliECx6k7pbzFRycVrgxvB+BFZ2ptxqCCCxAaVqzSJXz?=
	=?iso-8859-1?q?jD00C6v/xAFwEpd2CkUOKbWCit5zOCvFZor6MxxWBdkqf5fBdqc0ng9Eggds?=
	=?iso-8859-1?q?3VH+vbNz7U2AxhqCwTod8fZgQraKVqOBdGd+eGnzPB6NjavDq3MAE/Ih2A1S?=
	=?iso-8859-1?q?NhlF0nX3Lp3iP5frnup8gm41gjuW3GGvh4slNhYd1rX+O3L0c7FbFYSQMifm?=
	=?iso-8859-1?q?yBpVxEBiVKbRBwaJqwDdFLXk0h/gEBK6wNtx0DI/zKZ8L0qlbHYYI/uER95X?=
	=?iso-8859-1?q?QxIeXy+xpad+LrOZSlYIm7w4L7RgrpPbEghk+YzpkQtwtCSbAI49043d9h3t?=
	=?iso-8859-1?q?q7FCZ07g1bseZH/zfdJq7cGEe/4f7McRxR9Ip0Xzt52jWm2MiQCGK58b4DLe?=
	=?iso-8859-1?q?x/4zTfEbpkgF8lMGVRetYRIacsTEMDvmmzZRBK3UN/Jvuw6uSHVONyGKYce+?=
	=?iso-8859-1?q?xHd0gC1+/bN2JEb8xAdvWlQprcosY/FTvT35+2g=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 53359466-70b5-4bf3-a057-08da9040ed39
X-MS-Exchange-CrossTenant-originalarrivaltime: 06 Sep 2022 19:48:54.0220
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 Xf41walkvsG0lbpK6jt31wJKxU3yIJgDC2gKnnaDBvLTzFCj7iXRxLMyYqZKsRL1Sn0lJHkMQKldX2MyoL/CHQ==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SJ0PR10MB4670
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.895,Hydra:6.0.528,FMLib:17.11.122.1
 definitions=2022-09-06_09,2022-09-06_02,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 bulkscore=0
 mlxlogscore=999
 phishscore=0 mlxscore=0 spamscore=0 malwarescore=0 suspectscore=0
 adultscore=0 classifier=spam adjust=0 reason=mlx scancount=1
 engine=8.12.0-2207270000 definitions=main-2209060091
X-Proofpoint-ORIG-GUID: uJ7aQV3FSle5259LYBCTHSPyxl76J8Ut
X-Proofpoint-GUID: uJ7aQV3FSle5259LYBCTHSPyxl76J8Ut
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1662493798;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=FuNmZU+UMH7Y8mw5oBbNoLDQmSkii/s9w06bWHod278=;
	b=5/t+/5N2Mab/8IcSwJsssOkW6h/vZhFvq1ZAR0j/NfnkJ3ObEiFAj4M08DFs1a45OtCoiM
	eYeFLK5QShp6SkGMZKDSQt79+mFl9U+KE5Ybjrf+IMMC+zRUt+Il+2BosZoU++1N97mBMr
	IVbaX7r6xce8VeRVyFpC7IPajtHavF8=
ARC-Authentication-Results: i=2;
	imf14.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=hJAvIoXo;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b="BP/FjB1B";
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=pass (imf14.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1662493798; a=rsa-sha256;
	cv=pass;
	b=hid+u8mNtQLPd+9weYeFE+MZ1vmY5rDSB6PAvu8p3DrEiipgApLOkEOTzVboTyBWCrTIdc
	0K9oaMJttC0ypKUklO/6YaLMomdD0RWy2KbU/Q9G3Eebmdr6BXt+ctLzMNKoXxQNC5OhTd
	/3EBbMsQQtJNNSTlBK96KBvc4JIu4ok=
X-Rspamd-Queue-Id: CF069100069
Authentication-Results: imf14.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=hJAvIoXo;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b="BP/FjB1B";
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=pass (imf14.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
X-Rspamd-Server: rspam02
X-Rspam-User: 
X-Stat-Signature: 4xq9u48mdzzbpctepu75uk9m31frwdtf
X-HE-Tag: 1662493798-90331
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Matthew Wilcox (Oracle)" <willy@infradead.org>

Use the VMA iterator instead.

Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Reviewed-by: Vlastimil Babka <vbabka@suse.cz>
Reviewed-by: Davidlohr Bueso <dave@stgolabs.net>
---
 arch/powerpc/kernel/vdso.c              |  6 +++---
 arch/powerpc/mm/book3s32/tlb.c          | 11 ++++++-----
 arch/powerpc/mm/book3s64/subpage_prot.c | 13 ++-----------
 3 files changed, 11 insertions(+), 19 deletions(-)

diff --git a/arch/powerpc/kernel/vdso.c b/arch/powerpc/kernel/vdso.c
index 0da287544054..94a8fa5017c3 100644
--- a/arch/powerpc/kernel/vdso.c
+++ b/arch/powerpc/kernel/vdso.c
@@ -113,18 +113,18 @@ struct vdso_data *arch_get_vdso_data(void *vvar_page)
 int vdso_join_timens(struct task_struct *task, struct time_namespace *ns)
 {
 	struct mm_struct *mm = task->mm;
+	VMA_ITERATOR(vmi, mm, 0);
 	struct vm_area_struct *vma;
 
 	mmap_read_lock(mm);
-
-	for (vma = mm->mmap; vma; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		unsigned long size = vma->vm_end - vma->vm_start;
 
 		if (vma_is_special_mapping(vma, &vvar_spec))
 			zap_page_range(vma, vma->vm_start, size);
 	}
-
 	mmap_read_unlock(mm);
+
 	return 0;
 }
 
diff --git a/arch/powerpc/mm/book3s32/tlb.c b/arch/powerpc/mm/book3s32/tlb.c
index 19f0ef950d77..9ad6b56bfec9 100644
--- a/arch/powerpc/mm/book3s32/tlb.c
+++ b/arch/powerpc/mm/book3s32/tlb.c
@@ -81,14 +81,15 @@ EXPORT_SYMBOL(hash__flush_range);
 void hash__flush_tlb_mm(struct mm_struct *mm)
 {
 	struct vm_area_struct *mp;
+	VMA_ITERATOR(vmi, mm, 0);
 
 	/*
-	 * It is safe to go down the mm's list of vmas when called
-	 * from dup_mmap, holding mmap_lock.  It would also be safe from
-	 * unmap_region or exit_mmap, but not from vmtruncate on SMP -
-	 * but it seems dup_mmap is the only SMP case which gets here.
+	 * It is safe to iterate the vmas when called from dup_mmap,
+	 * holding mmap_lock.  It would also be safe from unmap_region
+	 * or exit_mmap, but not from vmtruncate on SMP - but it seems
+	 * dup_mmap is the only SMP case which gets here.
 	 */
-	for (mp = mm->mmap; mp != NULL; mp = mp->vm_next)
+	for_each_vma(vmi, mp)
 		hash__flush_range(mp->vm_mm, mp->vm_start, mp->vm_end);
 }
 EXPORT_SYMBOL(hash__flush_tlb_mm);
diff --git a/arch/powerpc/mm/book3s64/subpage_prot.c b/arch/powerpc/mm/book3s64/subpage_prot.c
index 60c6ea16a972..d73b3b4176e8 100644
--- a/arch/powerpc/mm/book3s64/subpage_prot.c
+++ b/arch/powerpc/mm/book3s64/subpage_prot.c
@@ -149,24 +149,15 @@ static void subpage_mark_vma_nohuge(struct mm_struct *mm, unsigned long addr,
 				    unsigned long len)
 {
 	struct vm_area_struct *vma;
+	VMA_ITERATOR(vmi, mm, addr);
 
 	/*
 	 * We don't try too hard, we just mark all the vma in that range
 	 * VM_NOHUGEPAGE and split them.
 	 */
-	vma = find_vma(mm, addr);
-	/*
-	 * If the range is in unmapped range, just return
-	 */
-	if (vma && ((addr + len) <= vma->vm_start))
-		return;
-
-	while (vma) {
-		if (vma->vm_start >= (addr + len))
-			break;
+	for_each_vma_range(vmi, vma, addr + len) {
 		vma->vm_flags |= VM_NOHUGEPAGE;
 		walk_page_vma(vma, &subpage_walk_ops, NULL);
-		vma = vma->vm_next;
 	}
 }
 #else

